test/integration/revocation_test.go

```diff
@@ -59,134 +180,166 @@ func TestRevocation(t *testing.T) {
+	var eMu sync.Mutex
+	var wg sync.WaitGroup
+
+	for _, kind := range []certKind{precert, finalcert} {
+		for _, reason := range []int{ocsp.Unspecified, ocsp.KeyCompromise} {
+			for _, method := range []authMethod{byAccount, byAuth, byKey} {
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					cert := issueAndRevoke(testCase{
+						method: method,
+						reason: reason,
+						kind:   kind,
+						// We do not expect any of these revocation requests to error.
+						// The ones done byAccount will succeed as requested, but will not
+						// result in the key being blocked for future issuance.
+						// The ones done byAuth will succeed, but will be overwritten to have
+						// reason code 5 (cessationOfOperation).
+						// The ones done byKey will succeed, but will be overwritten to have
+						// reason code 1 (keyCompromise), and will block the key.
+					})
+
+					// If the request was made by demonstrating control over the
+					// names, the reason should be overwritten to CessationOfOperation (5),
+					// and if the request was made by key, then the reason should be set to
+					// KeyCompromise (1).
+					expectedReason := reason
+					switch method {
+					case byAuth:
+						expectedReason = ocsp.CessationOfOperation
+					case byKey:
+						expectedReason = ocsp.KeyCompromise
+					default:
+					}
+
+					eMu.Lock()
+					expectations = append(expectations, expectation{
+						Certificate:      cert,
+						name:             fmt.Sprintf("%s_%d_%s", kind, reason, method),
+						revocationReason: expectedReason,
+					})
+					eMu.Unlock()
+				}()
 			}
+		}
+	}
 
-			// Revoke the cert using the specified key and client.
-			err = revokeClient.RevokeCertificate(
-				revokeClient.Account,
-				cert,
-				revokeKey,
-				tc.reason,
-			)
+	wg.Wait()
 
-			test.AssertNotError(t, err, "revocation should have succeeded")
+	runUpdater(t, path.Join(os.Getenv("BOULDER_CONFIG_DIR"), "crl-updater.json"))
+	revocations := getAllCRLs(t)
 
-			// Check the OCSP response for the certificate again. It should now be
-			// revoked. If the request was made by demonstrating control over the
-			// names, the reason should be overwritten to CessationOfOperation (5),
-			// and if the request was made by key, then the reason should be set to
-			// KeyCompromise (1).
-			ocspConfig = ocsp_helper.DefaultConfig.WithExpectStatus(ocsp.Revoked)
-			switch tc.method {
-			case byAuth:
-				ocspConfig = ocspConfig.WithExpectReason(ocsp.CessationOfOperation)
-			case byKey:
-				ocspConfig = ocspConfig.WithExpectReason(ocsp.KeyCompromise)
-			default:
-				ocspConfig = ocspConfig.WithExpectReason(tc.reason)
-			}
-			_, err = ocsp_helper.ReqDER(cert.Raw, ocspConfig)
+	for _, expectation := range expectations {
+		t.Run(expectation.name, func(t *testing.T) {
+			t.Parallel()
+			_, err := ocsp_helper.ReqDER(expectation.Raw, ocsp_helper.
+				DefaultConfig.
+				WithExpectStatus(ocsp.Revoked).
+				WithExpectReason(expectation.revocationReason))
```

Comment:
This might be the only place in the whole codebase where we break chained method calls onto separate lines. I don't love this style (my brain expects trailing punctuation to be a comma), and it violates the Google Go style guide's exhortation to ["Avoid introducing a line break if it would align the rest of the line with an indented code block"](https://google.github.io/styleguide/go/decisions#indentation-confusion), since these line up with the `test.AssertNotError` below them, and that ["function and method calls should not be separated based solely on line length."](https://google.github.io/styleguide/go/decisions#function-formatting)

I'd prefer to simply have a line that is a little bit too long, either:
```suggestion
			_, err := ocsp_helper.ReqDER(
				expectation.Raw, ocsp_helper.DefaultConfig.WithExpectStatus(ocsp.Revoked).WithExpectReason(expectation.revocationReason))
```
or
```suggestion
			_, err := ocsp_helper.ReqDER(expectation.Raw, ocsp_helper.DefaultConfig.WithExpectStatus(ocsp.Revoked).WithExpectReason(expectation.revocationReason))
```

Reply:
Done 👍🏻